authentik/events/signals.py

```diff
@@ -64,6 +68,26 @@ def on_user_write(sender, request: HttpRequest, user: User, data: dict[str, Any]
     data["created"] = kwargs.get("created", False)
     Event.new(EventAction.USER_WRITE, **data).from_http(request, user=user)
 
+    """Preserve the user's old password if UniquePasswordPolicy is enabled anywhere"""
```

Comment:
This should really listen for a `password_changed` signal, but we don't have access to the request in there :/

Reply:
Moved to the UniquePasswordPolicy app :+1: